include/gaussian.h

```diff
@@ -73,21 +74,44 @@ class Gaussian
     std::shared_ptr<Ellipse> _ellipse;
     std::shared_ptr<GaussianIntegral> _integral;
 
+    template <typename T>
+    void _set_if_not_nullptr(std::shared_ptr<T> x, std::shared_ptr<T> to_set, std::string name) {
+        if(to_set == nullptr) throw std::invalid_argument(this->str() + "Can't set " + name + " to nullptr");
+        x = std::move(to_set);
```

Comment:
I don't see how this can do anything.  `x` and `to_set` are both passed by value, so you're moving out of a temporary `shared_ptr` into another temporary `shared_ptr` and not returning anything; the objects the caller passed in will be unaffected.